lib/src/lifecycle_module.dart

```diff
@@ -321,11 +321,19 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
     }
 
     _state = LifecycleState.loading;
-    _transition = new Completer<Null>();
 
-    _load().then(_transition.complete).catchError(_transition.completeError);
+    // Keep track of this load's completer
+    final transition = new Completer<Null>();
 
-    return _transition.future;
+    // because this one can get overwritten
+    _transition = transition;
+
+    _load().then(transition.complete).catchError((error, trace) {
+      transition.completeError(error, trace);
+      _state = LifecycleState.loaded;
```

Comment:
For suspend and resume, if the transition fails, the state is reverted. In other words..

- `loaded` to `isSuspending` back to `loaded`
- `suspended` to `isResuming` back to `suspended`

But for load and unload, we are forcing through to the expected state despite the failure..
- `initialized` to `isLoading` to `loaded`, despite `onLoad` failure
- `loaded` to `isUnloading` to `unloaded`, despite `onUnload` failure

It seems a bit odd that we would have differing (in fact, opposite) behaviors for these two sets